conda_env/cli/main_create.py

```diff
@@ -22,6 +23,7 @@
     conda env create -n=foo
     conda env create -f=/path/to/environment.yml
     conda env create --name=foo --file=environment.yml
+    conda env create --name=foo --binstar=username/foo
```

Comment:
This syntax is incorrect -- using it we would have to add `--github` or `--gist` or `--whatever` for every single remote location we want to support.  There are two parts to this, adding the concept of an Environment Spec and then the backend to lookup those specs against known sources.  I'm going to break it out into an issue on its own, then reference this PR.